lib/fabrik/DirectorInstance.js

```diff
@@ -5,21 +5,38 @@ const Promise = require('bluebird');
 const _ = require('lodash');
 const yaml = require('js-yaml');
 const BaseInstance = require('./BaseInstance');
+const CfPlatformManager = require('./CfPlatformManager');
+const K8sPlatformManager = require('./K8sPlatformManager');
 const config = require('../config');
 const logger = require('../logger');
 const errors = require('../errors');
 const jwt = require('../jwt');
 const utils = require('../utils');
 const NotFound = errors.NotFound;
+const BadRequest = errors.BadRequest;
 const ServiceInstanceNotFound = errors.ServiceInstanceNotFound;
 const SecurityGroupNotCreated = errors.SecurityGroupNotCreated;
 const ScheduleManager = require('../jobs');
 const CONST = require('../constants');
 
 class DirectorInstance extends BaseInstance {
-  constructor(guid, manager) {
+  constructor(guid, manager, platformManager) {
     super(guid, manager);
     this.networkSegmentIndex = undefined;
+    this.platformManager = !platformManager ? undefined : platformManager;
+  }
+
+  assignPlatformManager() {
```

Comment:
This method should be part of BaseController IMO. The code 

```
return this.inspectContainer()
 .then(() => JSON.parse(this.getEnvironment().context))
```
and

```
this.manager.director.getDeploymentNameForInstanceId(this.guid)
      .then(deploymentName => this.manager.director.getDeploymentProperty(deploymentName, 'platform-context'))

```

can be part of specific instance methods which can be called from there.

Reply:
PlatformManager instantiation might be changed.